src/gui/SubWindow.cpp

```diff
@@ -111,6 +111,13 @@ SubWindow::SubWindow(QWidget *parent, Qt::WindowFlags windowFlags) :
  */
 void SubWindow::paintEvent( QPaintEvent * )
 {
+	// Don't paint any of the other stuff if the sub window is maximized
```

Comment:
The `p.fillRect( rect, isActive ? activeColor() : p.pen().brush() );` does not look like "other stuff", but rather like background which should also be required on maximized windows? Can you please elaborate why this is not needed for maximized windows?

Reply:
The `rect` is the rectangle of the title bar which should not be rendered when the window is maximized. It is defined a few lines before the `fillRect` call:
```
QRect rect( 0, 0, width(), m_titleBarHeight );
```

Painting it when the window is maximized led to the following buggy rendering where the title bar is rendered behind the main content of the sub window:

![7512-ArtifactsForMaximizedInstrumentWindow](https://github.com/user-attachments/assets/30ed0d3b-475b-4e31-a3a6-96c426d88fdc)

When the window is maximized we only want to render the child content and therefore we exit early if the window is maximized.

Reply:
Oh, of course, that rectangle. Thanks, resolving.